azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,326 @@
+package videoanalyzer
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/videoanalyzer/mgmt/2021-05-01-preview/videoanalyzer"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	msiparse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/msi/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/msi/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/videoanalyzer/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceVideoAnalyzer() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceVideoAnalyzerCreateUpdate,
+		Read:   resourceVideoAnalyzerRead,
+		Update: resourceVideoAnalyzerCreateUpdate,
+		Delete: resourceVideoAnalyzerDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.VideoAnalyzerID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-z0-9]{3,24}$"),
+					"Video Analyzer name must be 3 - 24 characters long, contain only lowercase letters and numbers.",
+				),
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"storage_account": {
+				Type:     pluginsdk.TypeSet,
```

Comment:
@EliiseS I am not sure because the linter error is still here?! And as far as I know it should be `pluginsdk.TypeList` when you use `MaxItems: 1`, but it is really only to the best of my knowledge...

Reply:
All the build passed, so there's no error as far as I can tell

Reply:
If I scroll here (https://github.com/terraform-providers/terraform-provider-azurerm/pull/12665/files) to the `video_analyzer_resource.go` file I see 
```
Check failure on line 59 in azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

GitHub Actions
/ tflint
azurerm/internal/services/videoanalyzer/video_analyzer_resource.go#L59
S018: schema should use TypeList with MaxItems 1
```

but I also do not know why this is really mandatory - would need to look into S018 :)

Reply:
Oh interesting! I'll def take a look at it tomorrow then, odd that it doesn't show up elsewhere 😱 Thanks for pointing it out!

Reply:
I suspect this check is because there's no reason to use a set with a single item when a list has better usability

Reply:
I've gone with using a list now :)